components/ntp_sponsored_images/browser/ntp_sponsored_images_service.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_NTP_SPONSORED_IMAGES_BROWSER_NTP_SPONSORED_IMAGES_SERVICE_H_
+#define BRAVE_COMPONENTS_NTP_SPONSORED_IMAGES_BROWSER_NTP_SPONSORED_IMAGES_SERVICE_H_
+
+#include "brave/components/ntp_sponsored_images/browser/ntp_sponsored_images_component_manager.h"
+#include "components/keyed_service/core/keyed_service.h"
+
+namespace content {
+class BrowserContext;
+}  // namespace content
+
+class NTPSponsoredImagesService
+    : public KeyedService,
+      public NTPSponsoredImagesComponentManager::Observer {
+ public:
+  NTPSponsoredImagesService(content::BrowserContext* browser_context,
+                            NTPSponsoredImagesComponentManager* manager);
+  ~NTPSponsoredImagesService() override;
+
+  NTPSponsoredImagesService(const NTPSponsoredImagesService&) = delete;
```

Comment:
use `DISALLOW_COPY_AND_ASSIGN`

Reply:
OK to leave as-is, IMO. See note at https://github.com/brave/brave-core/pull/4075/files/2f7f51f8a0ffee135d823311f7bb464dd74e6acc#r370352211

Reply:
I'm a bit torn on that change, but I think for now either way is fine